eth/db/header.py

```diff
@@ -149,43 +161,60 @@ def header_exists(self, block_hash: Hash32) -> bool:
     def persist_header(self,
                        header: BlockHeader
                        ) -> Tuple[Tuple[BlockHeader, ...], Tuple[BlockHeader, ...]]:
+        return self.persist_header_chain((header,))
+
+    def persist_header_chain(self,
+                             headers: Iterable[BlockHeader]
+                             ) -> Tuple[Tuple[BlockHeader, ...], Tuple[BlockHeader, ...]]:
         """
-        :returns: iterable of headers newly on the canonical chain
+        Return two iterable of headers, the first containing the new canonical headers,
+        the second containing the old canonical headers
         """
-        if header.parent_hash != GENESIS_PARENT_HASH:
-            try:
-                self.get_block_header_by_hash(header.parent_hash)
-            except HeaderNotFound:
+
+        try:
+            first_header = first(headers)
+        except StopIteration:
+            return tuple(), tuple()
+        else:
+
+            consecutive_pairs = sliding_window(2, headers)
+            if any(parent.hash != child.parent_hash for parent, child in consecutive_pairs):
+                raise ValidationError("Invalid headers. Headers need to be contiguous")
```

Comment:
Would be nice to print the consecutive parent and child headers, and the parent_hash that the child expected.

Reply:
Yes, I rushed this. Had to decide about finishing this properly vs letting my 6yo waiting in front of the school in tears. Tough decision. I'll finish this before merging.